### PR TITLE
Fix Examples/{UIExplorer,Movies}

### PR DESCRIPTION
**Pull request by leeight**
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request:

(You can skip this if you're fixing a typo or adding an app to the Showcase.)

UIExplorerActivity.java and MoviesActivity.java should override `getReactNativeHost` method. And this PR will fix https://github.com/facebook/react-native/issues/8215.

**Test plan (required)**

Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

Make sure tests pass on both Travis and Circle CI.

**Code formatting**

Look around. Match the style of the rest of the codebase. See also the simple [style guide](https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md#style-guide).

For more info, see the ["Pull Requests" section of our "Contributing" guidelines](https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md#pull-requests).

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
By analyzing the blame information on this pull request, we identified @caabernathy and @janicduplessis to be potential reviewers.

**Comment by satya164** [comment]
Hey! Thanks for the PR. Can you update the code to follow similar structure to the generator? You'll need to add a `UIExplorerApp` class and a `UIExplorerActivity` class, e.g. - https://github.com/facebook/react-native/blob/master/local-cli/generator-android/templates/package/MainApplication.java and https://github.com/facebook/react-native/blob/master/local-cli/generator-android/templates/package/MainActivity.java

File: ReactAndroid/src/main/java/com/facebook/react/ReactActivity.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ protected ReactNativeHost getReactNativeHost() {
    * determining this (default just checks {@code BuildConfig}).
    */
   protected boolean getUseDeveloperSupport() {
-    return ((ReactApplication) getApplication()).getReactNativeHost().getUseDeveloperSupport();
+    return getReactNativeHost().getUseDeveloperSupport();
```
Comment (satya164) [inline ReactAndroid/src/main/java/com/facebook/react/ReactActivity.java]: You won't need this change if you follow the template.

Comment (leeight) [inline ReactAndroid/src/main/java/com/facebook/react/ReactActivity.java]: @satya164 I submit a new commit which added UIExplorerApplication and MoviesApplication, please review it. Thanks

File: Examples/Movies/android/app/src/main/java/com/facebook/react/movies/MoviesActivity.java [hunk 2]
```diff
@@ -1,9 +1,9 @@
 /**
  * The examples provided by Facebook are for non-commercial testing and
  * evaluation purposes only.
- *
+ * <p/>
```
Comment (satya164) [inline Examples/Movies/android/app/src/main/java/com/facebook/react/movies/MoviesActivity.java]: added by mistake?

File: Examples/Movies/android/app/src/main/java/com/facebook/react/movies/MoviesActivity.java [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
  * Facebook reserves all rights not expressly granted.
- *
+ * <p/>
```
Comment (satya164) [inline Examples/Movies/android/app/src/main/java/com/facebook/react/movies/MoviesActivity.java]: remove this please

File: Examples/UIExplorer/android/app/src/main/java/com/facebook/react/uiapp/UIExplorerApplication.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+package com.facebook.react.uiapp;
```
Comment (satya164) [inline Examples/UIExplorer/android/app/src/main/java/com/facebook/react/uiapp/UIExplorerApplication.java]: Add the license header (same as `UIExplorerActivity.java`)

File: Examples/Movies/android/app/src/main/java/com/facebook/react/movies/MoviesApplication.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package com.facebook.react.movies;
```
Comment (satya164) [inline Examples/Movies/android/app/src/main/java/com/facebook/react/movies/MoviesApplication.java]: Add the license header (same as `MoviesActivity.java`)

**Comment by satya164** [comment]
Looks good, just the few things I mentioned and then we can merge.

File: Examples/Movies/android/app/src/main/java/com/facebook/react/movies/MoviesActivity.java (hunk 2 above)
Comment (leeight) [inline Examples/Movies/android/app/src/main/java/com/facebook/react/movies/MoviesActivity.java]: Auto Formatting issues, i'll fix it.

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**